dissect/esedb/record.py

```diff
@@ -183,6 +184,48 @@ def get(self, column: Column, raw: bool = False) -> RecordValue:
         if value is not None:
             return self._parse_value(column, value, tag_field)
 
+    def serialize(self) -> dict[str, RecordValue]:
+        """Serialize the record columns as a dictionnary."""
+        obj = dict()
+        for idx in range(self._last_fixed_id):
+            try:
+                column = self.table._column_id_map[idx]
+            except:
+                continue
+            value = self._get_fixed(column)
+            try:
+                obj[column.name] = self._parse_value(column, value)
+            except Exception as e:
+                obj[column.name] = f"!ERROR! [fixed] {e}"
+
+        for idx in range(128, self._last_variable_id):
+            try:
+                column = self.table._column_id_map[idx]
+            except:
+                continue
+            value = self._get_variable(column)
+
+            try:
+                obj[column.name] = self._parse_value(column, value)
+            except Exception as e:
+                obj[column.name] = f"!ERROR! [variable] {e}"
+
+        for idx in range(self._tagged_data_count):
+            try:
+                tag_field = self._get_tag_field(idx)
+                column = self.table._column_id_map[tag_field.identifier]
+            except:
+                continue
+
+            tag_field, value = self._get_tagged(column)
+
+            try:
+                obj[column.name] = self._parse_value(column, value, tag_field)
+            except Exception as e:
+                obj[column.name] = f"!ERROR! [tag] {e}"
+
+        return obj
```

Comment:
```suggestion
        obj = {}

        def _iter_columns() -> Iterator[Column]:
            for column_id in itertools.chain(range(self, _last_fixed_id), range(128, self._last_variable_id)):
                yield self.table._column_id_map[column_id]

            for idx in range(self._tagged_data_count):
                tag_field = self._get_tag_field(idx)
                yield self.table._column_id_map[tag_field.identifier]

        for column in _iter_columns():
            try:
                obj[column.name] = self.get(column, raw)
            except Exception as e:
                obj[column.name] = f"!ERROR! {e}"

        return obj
```

I didn't test it but would this work too? Also need to add the `itertools` import of course.

Reply:
Tested, it works!